### PR TITLE
ILR dev branch: Merge in BC learning rate scheduler option

### PR DESCRIPTION
**Pull request by shwang**

### CODE REVIEW

File: tests/test_dagger.py [hunk 1]
```diff
@@ -66,7 +66,6 @@ def make_trainer(tmpdir, beta_schedule=dagger.LinearBetaSchedule(1)):
         env,
         tmpdir,
         beta_schedule,
-        optimizer_kwargs=dict(lr=1e-3),
```
Comment (shwang) [inline tests/test_dagger.py:69]: This shouldn't be here. I will remove before merge.

**Comment by shwang** [comment]
Lint passes fine and all the `test_bc` tests pass fine on my computer, so I believe that this PR should be fine, testing wise.

File: src/imitation/algorithms/bc.py [hunk 2]
```diff
@@ -168,16 +184,20 @@ def __init__(
         observation_space: gym.Space,
         action_space: gym.Space,
         *,
-        policy_class: Type[policies.BasePolicy] = base.FeedForward32Policy,
+        policy_class: Type[policies.ActorCriticPolicy] = base.FeedForward32Policy,
```
Comment (qxcv) [inline src/imitation/algorithms/bc.py:187]: Maybe add a comment explaining why this is needed?

File: src/imitation/algorithms/bc.py [hunk 3]
```diff
@@ -209,20 +231,36 @@ def __init__(
         self.observation_space = observation_space
         self.policy_class = policy_class
         self.device = device = utils.get_device(device)
+
+        # SB3's `ActorCriticPolicy` automatically initializes a optimizer which requires
+        # an argument of this type, but we don't use.
+        unused_lr_schedule = ConstantLRSchedule()
```
Comment (qxcv) [inline src/imitation/algorithms/bc.py:237]: Maybe you should set the LR to something ridiculous (like `None`, or maybe `NaN`) to make errors more obvious?

**Review by qxcv — APPROVED** [review]
Some minor suggestions, but otherwise LGTM.

**Pull request merged**